pkg/api/types.go

```diff
@@ -44,13 +44,22 @@ type DeschedulerStrategy struct {
 	Params *StrategyParameters
 }
 
+// Namespaces carries a list of included/excluded namespaces
+// for which a given strategy is applicable
+type Namespaces struct {
+	Include []string
+	Exclude []string
+}
+
 // Only one of its members may be specified
 type StrategyParameters struct {
 	NodeResourceUtilizationThresholds *NodeResourceUtilizationThresholds
-	NodeAffinityType                  []string
-	PodsHavingTooManyRestarts         *PodsHavingTooManyRestarts
-	MaxPodLifeTimeSeconds             *uint
+	NodeAffinity                      *NodeAffinity
+	PodsHavingTooManyRestarts         []PodsHavingTooManyRestarts
+	PodLifeTime                       []PodLifeTime
```

Comment:
@ingvagabund pointed out to me that this change (as I recommended in https://github.com/kubernetes-sigs/descheduler/pull/338#discussion_r454429921, adding `Namespaces` to each strategy's struct) would break the current config API since `NodeAffinityType` and `MaxPodLifeTimeSeconds` don't currently have their own structs.

So to avoid changing the API right at the end of a release, I take back what I said and think adding `Namespaces` to `StrategyParameters`, to get Namespace filtering working as planned for 1.19, is the simplest approach for now.

There have been a lot of ideas for how to change the api, so I would prefer to gather those up and target a rehaul in 1.20, where perhaps `Namespaces` can be set on each strategy's struct (or something similar)